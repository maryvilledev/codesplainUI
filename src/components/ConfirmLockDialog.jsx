import React, { PropTypes } from 'react';

import Dialog from 'material-ui/Dialog';
import FlatButton from 'material-ui/FlatButton';

const ConfirmLockDialog = ({ accept, isOpen, reject }) => {
  const actionButtons = [
    <FlatButton
      label="No"
      secondary
      onTouchTap={reject}
    />,
    <FlatButton
      label="Yes"
      primary
      onTouchTap={accept}
    />,
  ];
  return (
    <Dialog
      actions={actionButtons}
      modal
      open={isOpen}
      title="Are you sure you want to lock editing?"
    >
      Note that you will not be able to revert back to edit mode
    </Dialog>
  );
};

ConfirmLockDialog.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  accept: PropTypes.func.isRequired,
  reject: PropTypes.func.isRequired,
};

export default ConfirmLockDialog;
