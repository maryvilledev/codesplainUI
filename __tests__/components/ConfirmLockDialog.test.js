import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import ConfirmLockDialog from '../../src/components/ConfirmLockDialog';

const mockFunctionProps = {
  accept: jest.fn(),
  reject: jest.fn(),
};

describe('<ConfirmLockDialog />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot when it is open', () => {
    const wrapper = shallowWithContext(
      <ConfirmLockDialog
        isOpen={true}
        {...mockFunctionProps}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when it is closed', () => {
    const wrapper = shallowWithContext(
      <ConfirmLockDialog
        isOpen={false}
        {...mockFunctionProps}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
