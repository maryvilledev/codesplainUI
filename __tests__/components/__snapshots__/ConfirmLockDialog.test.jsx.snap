exports[`<ConfirmLockDialog /> matches snapshot when it is closed 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="No"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={false}
        secondary={true} />,
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="Yes"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false} />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={false}
  repositionOnUpdate={true}
  title="Are you sure you want to lock editing?">
  Note that you will not be able to revert back to edit mode
</Dialog>
`;

exports[`<ConfirmLockDialog /> matches snapshot when it is open 1`] = `
<Dialog
  actions={
    Array [
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="No"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={false}
        secondary={true} />,
      <FlatButton
        disabled={false}
        fullWidth={false}
        label="Yes"
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={true}
        secondary={false} />,
    ]
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={true}
  open={true}
  repositionOnUpdate={true}
  title="Are you sure you want to lock editing?">
  Note that you will not be able to revert back to edit mode
</Dialog>
`;
